### PR TITLE
Reflect Emoji Skin Tone automatically in Emoji List.

### LINKED ISSUE #101: Reflect Emoji Skin Tone automatically in Emoji List
When changing the skin tone of an Emoji in a Category the Emoji Icon in the list should be updated. 

1. Look for 👊
2. Long press on 👊 
3. Change in Dialog to 👊🏻
4. In the list 👊🏻 should be displayed now instead of 👊

If we do this we should no longer then display 👊 & 👊🏻 in the recent Emojis. Instead just display the one that was used last.

### PR DESCRIPTION
**Pull request by vanniktech**
Partially taken from #149. Some things are still left to do:

- [x] Still a little lag between showing the new Emoji and the old one
- [x] Some skin modifiers are not working properly (for some reason some of the guys that are only 4 characters long e.g. 270a)
- [x] Tests

Fixes #101

### CODE REVIEW

File: emoji/src/main/java/com/vanniktech/emoji/EmojiImageView.java [hunk 1]
```diff
@@ -66,9 +71,65 @@ public EmojiImageView(final Context context, final AttributeSet attrs) {
...
+          }
+
+          return false;
+        }
+      });
+    } else {
+      setOnLongClickListener(null);
+    }
+
+    ImageLoadingTask task = (ImageLoadingTask) getTag();
```
Comment (rubengees) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiImageView.java]: Maybe avoid saving the `task` in the tag altogether and just call `dispose` on the `Disposable` class member?

File: emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import android.content.SharedPreferences;
+import android.support.annotation.NonNull;
+import com.vanniktech.emoji.emoji.Emoji;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+final class RecentVariantManager { // TODO: test it
```
Comment (rubengees) [inline emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java]: Make this an `interface` like the `RecentEmojiManger` so consumers could choose to implement persistance themselfs?

File: emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    //}
+
+    // TODO: sometimes those 4 characters are not working (e.g. 270a)
+
+    for (final Emoji recentEmoji : recentVariantsList) {
+      if (recentEmoji.getBase().equals(newVariantBase)) {
+        if (recentEmoji.equals(newVariant)) { // Same skin-tone was used.
+          return;
+        } else {
+          recentVariantsList.remove(recentEmoji);
```
Comment (rubengees) [inline emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java]: This is potentially slower than possible.

`remove` needs to find the `recentEmoji` again, by iteration the list, even though we have already found it.
When using `fori` in which we memorize the index, we can avoid that.

**Comment by rubengees** [comment]
The lag does not look too bad, WhatsApp also has it if you look closely.

File: emoji/src/main/java/com/vanniktech/emoji/EmojiImageView.java (hunk 1 above)
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiImageView.java]: I think the Task is only necessary when at once trying to display all the Emojis. I don't think we need it to just switch the variant (We know for sure that you can only do this one by one). I think I'll just end up calling `setImageResource()` and also avoid the `setDrawable(null)` call which should get rid of the flickering.

File: emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java (hunk 2 above)
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java]: Thought about it and I don't think that you want to do this. If there's high demand for it we can still expose it. I prefer to keep as many internals internals.

File: emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java (hunk 3 above)
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/RecentVariantManager.java]: Yes definitely, it's also the implementation from #149 and I'm still going to optimize it but first I want to add some tests as the current implementation does seem to work (except for the edge case outlined above)

**Comment by vanniktech** [comment]
Yup also saw the lag there, I hope I can get rid of it, otherwise the functionality gain (not having to always re-select your favorite skin tone) outweighs the small lag that you can see.
